File: api/src/Entity/Objective.php
<?php

namespace App\Entity;

use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\ORM\Mapping as ORM;
use App\Entity\Base\BaseEntity;

/**
 * @ORM\Entity()
 * @ORM\Table(name="objectives")
 */
class Objective extends BaseEntity
{
    public function __construct()
    {
        $this->objective_subobjectives = new ArrayCollection();
        parent::__construct();
    }

    /**
     * @ORM\Id
     * @ORM\Column(type="string")
     * @ORM\GeneratedValue(strategy="UUID")
     */
    protected $objective_id;

    /**
     * @ORM\Column(type="string", nullable=true)
     */
    protected $objective_title;

    /**
     * @ORM\Column(type="text", nullable=true)
     */
    protected $objective_description;

    /**
     * @ORM\Column(type="smallint", nullable=true)
     */
    protected $objective_priority;

    /**
     * @ORM\ManyToOne(targetEntity="Exercise", inversedBy="exercise_objectives")
     * @ORM\JoinColumn(name="objective_exercise", referencedColumnName="exercise_id", onDelete="CASCADE")
     * @var Exercise
     */
    protected $objective_exercise;

    /**
     * @ORM\OneToMany(targetEntity="Subobjective", mappedBy="subobjective_objective")
     * @var Subobjective[]
     */
    protected $objective_subobjectives;

    public function getObjectiveId()
    {
        return $this->objective_id;
    }

    public function setObjectiveId($id)
    {
        $this->objective_id = $id;
        return $this;
    }

    public function getObjectiveTitle()
    {
        return $this->objective_title;
    }

    public function setObjectiveTitle($title)
    {
        $this->objective_title = $title;
        return $this;
    }

    public function getObjectiveDescription()
    {
        return $this->objective_description;
    }

    public function setObjectiveDescription($description)
    {
        $this->objective_description = $description;
        return $this;
    }

    public function getObjectivePriority()
    {
        return $this->objective_priority;
    }

    public function setObjectivePriority($priority)
    {
        $this->objective_priority = $priority;
        return $this;
    }

    public function getObjectiveExercise()
    {
        return $this->objective_exercise;
    }

    public function setObjectiveExercise($exercise)
    {
        $this->objective_exercise = $exercise;
        return $this;
    }

    public function getObjectiveSubobjectives()
    {
        return $this->objective_subobjectives;
    }

    public function setObjectiveSubobjectives($subobjectives)
    {
        $this->objective_subobjectives = $subobjectives;
        return $this;
    }
}

File: openex-player/src/main/java/io/openex/player/model/database/Objective.java
package io.openex.player.model.database;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import io.openex.player.helper.MonoModelDeserializer;
import io.openex.player.helper.MultiModelDeserializer;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "objectives")
public class Objective implements Base {
    @Id
    @Column(name = "objective_id")
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @JsonProperty("objective_id")
    private String id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "objective_exercise")
    @JsonSerialize(using = MonoModelDeserializer.class)
    @JsonProperty("objective_exercise")
    private Exercise exercise;

    @Column(name = "objective_title")
    @JsonProperty("objective_title")
    private String title;

    @Column(name = "objective_description")
    @JsonProperty("objective_description")
    private String description;

    @Column(name = "objective_priority")
    @JsonProperty("objective_priority")
    private Short priority;

    @OneToMany(mappedBy = "objective")
    @JsonSerialize(using = MultiModelDeserializer.class)
    @JsonProperty("objective_subobjectives")
    private List<SubObjective> subObjectives = new ArrayList<>();

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public Exercise getExercise() {
        return exercise;
    }

    public void setExercise(Exercise exercise) {
        this.exercise = exercise;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Short getPriority() {
        return priority;
    }

    public void setPriority(Short priority) {
        this.priority = priority;
    }

    public List<SubObjective> getSubObjectives() {
        return subObjectives;
    }

    public void setSubObjectives(List<SubObjective> subObjectives) {
        this.subObjectives = subObjectives;
    }
}
